# HG changeset patch
# User Rik <rik@octave.org>
# Date 1436321375 25200
#      Tue Jul 07 19:09:35 2015 -0700
# Branch stable
# Node ID 3300cb76cccdeaf47b9d8d589bd248e6c495778f
# Parent  2fc43288a6c48b8c02fa1ddcaf1d82249408dc7c
install.txi: Add warning about incorrect behavior of reference BLAS library.

* install.txi: Add warning about incorrect behavior of reference BLAS library.

diff --git a/doc/interpreter/install.txi b/doc/interpreter/install.txi
--- a/doc/interpreter/install.txi
+++ b/doc/interpreter/install.txi
@@ -185,20 +185,21 @@ files or the docstring of any Octave fun
 
 @node External Packages
 @subsection External Packages
 
 The following external packages are required:
 
 @table @asis
 @item BLAS
-Basic Linear Algebra Subroutine library
-(@url{http://www.netlib.org/blas}).  Accelerated @sc{blas} libraries such as
-ATLAS (@url{http://math-atlas.sourceforge.net}) are recommeded for
-better performance.
+Basic Linear Algebra Subroutine library.  Accelerated @sc{blas} libraries such
+as OpenBLAS (@url{http://www.openblas.net/}) or
+ATLAS (@url{http://math-atlas.sourceforge.net}) are recommended for best
+performance.  The reference implementation (@url{http://www.netlib.org/blas})
+is slow, unmaintained, and suffers from certain bugs in corner case inputs.
 
 @item LAPACK
 Linear Algebra Package (@url{http://www.netlib.org/lapack}).
 
 @item PCRE
 The Perl Compatible Regular Expression library (@url{http://www.pcre.org}).
 @end table
 
@@ -966,16 +967,37 @@ these files without @option{-g}.
 
 @item
 Some people have reported that calls to system() and the pager do not
 work on SunOS systems.  This is apparently due to having
 @w{@code{G_HAVE_SYS_WAIT}} defined to be 0 instead of 1 when compiling
 @code{libg++}.
 
 @item
+On systems where the reference @sc{blas} library is used the following
+matrix-by-vector multiplication incorrectly handles NaN values of the
+form @code{NaN * 0}.
+
+@example
+@group
+[NaN, 1; 0, 0] * [0; 1]
+@result{}
+[ 1
+  0 ]
+
+correct result @result{}
+[ NaN
+  0   ]
+@end group
+@end example
+
+Install a different @sc{blas} library such as OpenBLAS or ATLAS to correct
+this issue.
+
+@item
 On NeXT systems, linking to @file{libsys_s.a} may fail to resolve the
 following functions
 
 @example
 @group
 _tcgetattr
 _tcsetattr
 _tcflow
@@ -1027,13 +1049,13 @@ proper thing to do.
 If Octave is unable to find a header file because it is installed in a
 location that is not normally searched by the compiler, you can add the
 directory to the include search path by specifying (for example)
 @code{CPPFLAGS=-I/some/nonstandard/directory} as an argument to
 @code{configure}.  Other variables that can be specified this way are
 @env{CFLAGS}, @env{CXXFLAGS}, @env{FFLAGS}, and @env{LDFLAGS}.  Passing
 them as options to the configure script also records them in the
 @file{config.status} file.  By default, @env{CPPFLAGS} and @env{LDFLAGS}
-are empty, @env{CFLAGS} and @env{CXXFLAGS} are set to @qcode{"-g -O"} and
+are empty, @env{CFLAGS} and @env{CXXFLAGS} are set to @qcode{"-g -O2"} and
 @env{FFLAGS} is set to @qcode{"-O"}.
 
 @end itemize
 
